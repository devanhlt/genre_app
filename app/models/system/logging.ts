import { endOfDay, startOfDay, subDays } from "date-fns"
import { Instance, SnapshotIn, SnapshotOut, types } from "mobx-state-tree"

import { withSetPropAction } from "../helpers/withSetPropAction"

// const TypeId = types.model("TypeId", {
//   timestamp: types.optional(types.maybeNull(types.number), null),
//   date: types.optional(types.maybeNull(types.number), null),
// })

export const LoggingFilterModel = types.model("LoggingFilter", {
  keyword: types.optional(types.maybeNull(types.string), ""),
  pageNumber: types.optional(types.maybeNull(types.number), 1),
  pageSize: types.optional(types.maybeNull(types.number), 10),
  pageDraw: types.optional(types.maybeNull(types.number), 1),
  fromDate: types.optional(types.maybeNull(types.Date), startOfDay(subDays(new Date(), 1))),
  toDate: types.optional(types.maybeNull(types.Date), endOfDay(new Date())),
  status: types.optional(types.maybeNull(types.string), ""),
  system: types.optional(types.maybeNull(types.string), ""),
  logType: types.optional(types.maybeNull(types.string), ""),
})

export const LoggingModel = types
  .model("LoggingModel")
  .props({
    // id: types.optional(types.maybeNull(TypeId), null),
    id: types.optional(types.maybeNull(types.string), null),
    systemName: types.optional(types.maybeNull(types.string), null),
    apiReferal: types.optional(types.maybeNull(types.string), null),
    logType: types.optional(types.maybeNull(types.string), null),
    logCode: types.optional(types.maybeNull(types.number), null),
    logLevel: types.optional(types.maybeNull(types.number), null),
    logDetail: types.optional(types.maybeNull(types.string), null),
    method: types.optional(types.maybeNull(types.string), null),
    requestParam: types.optional(types.maybeNull(types.string), null),
    requestBody: types.optional(types.maybeNull(types.string), null),
    startTime: types.optional(types.maybeNull(types.string), null),
    endTime: types.optional(types.maybeNull(types.string), null),
    status: types.optional(types.maybeNull(types.string), null),
    createdBy: types.optional(types.maybeNull(types.string), null),
    createdDate: types.optional(types.maybeNull(types.number), null),
    updatedBy: types.optional(types.maybeNull(types.string), null),
    updatedDate: types.optional(types.maybeNull(types.number), null),
    idStr: types.optional(types.maybeNull(types.string), null),
    color: types.optional(types.maybeNull(types.string), null),
  })
  .actions(withSetPropAction)
  .views((logging) => ({
    get getLogging() {
      return logging
    },
  }))

export interface Logging extends Instance<typeof LoggingModel> {}
export interface LoggingSnapshotOut extends SnapshotOut<typeof LoggingModel> {}
export interface LoggingSnapshotIn extends SnapshotIn<typeof LoggingModel> {}
