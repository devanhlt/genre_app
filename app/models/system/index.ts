import { endOfDay, startOfDay, subDays } from "date-fns"
import { Instance, SnapshotOut, flow, getType, toGenerator, types } from "mobx-state-tree"

import { jsonToString } from "app/utils/helpers"
import { withEnvironment } from "../extensions/with-environment"
import { withRootStore } from "../helpers/withRootStore"
import { withSetPropAction } from "../helpers/withSetPropAction"
import { LoggingFilterModel, LoggingModel, LoggingSnapshotIn } from "./logging"
import SystemServices from "./services"
import { SystemModel } from "./system"
import SettingServices from "../setting/services"

const ALL_SYSTEM = {
  systemName: "ALL",
  baseUrl: null,
  adminEmail: null,
  adminPhone: null,
  totalLog: null,
  totalLogInfo: null,
  totalLogWarn: null,
  totalLogError: null,
  receiveLog: false,
}

export const SystemStoreModel = types
  .model("SystemStore")
  .props({
    systems: types.array(SystemModel),
    lstSystemLogging: types.array(LoggingModel),
    isLoadingSystem: false,
    loggingTotalCount: types.optional(types.number, 0),
    loggingRecordsFiltered: types.optional(types.number, 0),
    loggingRecordsTotal: types.optional(types.number, 0),
    loggingFilter: types.optional(LoggingFilterModel, {
      keyword: "",
      pageNumber: 1,
      pageSize: 10,
      pageDraw: 1,
      fromDate: startOfDay(subDays(new Date(), 1)),
      toDate: endOfDay(new Date()),
      status: "",
      logType: "",
      system: "ALL",
    }),
  })
  .extend(withEnvironment) // Extend environment
  .extend(withRootStore)
  .actions(withSetPropAction)
  .views((self) => {
    return {
      get systemsForList() {
        return self.systems
      },
      get systemsForListFilter() {
        return [ALL_SYSTEM].concat(self.systems)
      },
      get systemsLoggingForList() {
        return self.lstSystemLogging
      },

      get getLoggingCurrentFilter() {
        return {
          ...self.loggingFilter,
        }
      },
      get isEmptySystem() {
        return !self.systems.length
      },
    }
  })
  .actions((self) => ({
    fetchSystems: flow(function* fetchSystems() {
      self.setProp("isLoadingSystem", true)
      const systemServices = new SystemServices(self.rootStore)
      const response = yield* toGenerator(systemServices.getSystems())
      if (response.kind === "ok") {
        self.setProp("systems", response.systems)
      } else {
        console.tron.error(`Error fetching systems: ${jsonToString(response)}`, [])
      }
      self.setProp("isLoadingSystem", false)
    }),

    fetchLoggingSystems: flow(function* fetchLoggingSystems(
      loggingFiltering: Instance<typeof LoggingFilterModel>,
    ) {
      const currentFilter = { ...loggingFiltering }
      // Delete system param from the request params if choose all system
      if (currentFilter.system === ALL_SYSTEM.systemName) {
        delete currentFilter.system
      }
      const systemServices = new SystemServices(self.rootStore)
      const response = yield* toGenerator(systemServices.getLogging(currentFilter))

      if (response.kind === "ok") {
        self.setProp("lstSystemLogging", response.logging)
      } else {
        console.tron.error(`Error fetching systems: ${jsonToString(response)}`, [])
      }
    }),

    exportLogByFilter: flow(function* exportLogByFilter(
      loggingFiltering: Instance<typeof LoggingFilterModel>,
    ) {
      const systemServices = new SystemServices(self.rootStore)
      const response = yield* toGenerator(systemServices.exportLoggingByFilter(loggingFiltering))
      if (response.kind === "ok") {
        return response.logging
      } else {
        return []
      }
    }),

    exportSingleLogDetail: flow(function* exportLogByFilter(logging: LoggingSnapshotIn) {
      const systemServices = new SystemServices(self.rootStore)
      const response = yield* toGenerator(systemServices.exportSingleLogDetail(logging))
      if (response.kind === "ok") {
        return response.data
      } else {
        return ""
      }
    }),

    onChangeLoggingFilter: (loggingFiltering: Instance<typeof LoggingFilterModel>) => {
      self.setProp("loggingFilter", loggingFiltering)
    },

    onResetFilter: () => {
      self.setProp("loggingFilter", {
        keyword: "",
        pageNumber: 1,
        pageSize: 10,
        pageDraw: 1,
        fromDate: startOfDay(subDays(new Date(), 1)),
        toDate: endOfDay(new Date()),
        status: "",
        logType: "",
      })
    },

    /**
     * Update a system
     * Return true if success
     *
     */
    updateSystemConfig: flow(function* editSystemDetail(system: Instance<typeof SystemModel>) {
      const settingServices = new SettingServices(self.rootStore)
      const response = yield* toGenerator(settingServices.putSystemConfig(system))
      return response.kind === "ok" && response.success
    }),

    /**
     * Reset model
     */
    reset: flow(function* reset() {
      self.setProp("loggingFilter", {
        keyword: "",
        pageNumber: 1,
        pageSize: 10,
        pageDraw: 1,
        fromDate: startOfDay(subDays(new Date(), 1)),
        toDate: endOfDay(new Date()),
        status: "",
        logType: "",
      })
      self.setProp("systems", [])
      self.setProp("isLoadingSystem", false)
      self.setProp("loggingRecordsFiltered", 0)
      self.setProp("loggingRecordsTotal", 0)
      self.setProp("loggingTotalCount", 0)
    }),

    /**
     * Handle logout
     * Reset store
     */
    // Call when Instantiated
    afterCreate() {
      console.log("Instantiated " + getType(self).name)
    },
  }))

export interface SystemStore extends Instance<typeof SystemStoreModel> {}
export interface SystemStoreSnapshot extends SnapshotOut<typeof SystemStoreModel> {}
