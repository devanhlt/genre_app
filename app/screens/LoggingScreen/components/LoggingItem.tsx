import React from "react"
import { TextStyle, View, ViewStyle } from "react-native"

import { Typography } from "app/components"
import Badge from "app/components/Badge"
import { Divider } from "app/components/Divider"
import { Pressable } from "app/components/Pressable"
import { Logging } from "app/models/system/logging"
import { appColors, spacing } from "app/theme"
import { formatDate } from "app/utils/formatDate"
import { responsiveWidth } from "app/utils/screens"

export interface LoggingItemProps {
  item?: Logging
  onPressItem?: (item: Logging) => void
}

export default function LoggingItem({
  item,
  onPressItem = () => null,
}: LoggingItemProps): JSX.Element {
  const handlePressItem = () => {
    onPressItem?.(item)
  }

  if (!item) return null

  return (
    <Pressable style={$viewContainer} onPress={handlePressItem}>
      <View style={$logTypeView}>
        <Badge
          textColor={"#fff"}
          backgroundColor={item.color || appColors.common.bgRed}
          borderRadius={4}
        >
          {item.logType}
        </Badge>
      </View>
      <View style={$rowItem}>
        <Typography text="System:" preset="body04" style={$titleText} />
        <Divider size={spacing.size12} type="vertical" />
        <Typography text={item.systemName} preset="body04" style={$valueText} />
      </View>

      {/* <View style={$rowItem}>
        <Typography text="Message:" preset="body04" style={$titleText} />
        <Divider size={spacing.size12} type="vertical" />
        <Typography
          text={jsonToString(stringToJson(item?.logDetail || "{}"))}
          preset="body02"
          style={$valueText}
          numberOfLines={3}
        />
      </View> */}
      <View style={$rowItem}>
        <Typography text="Event time:" preset="body04" style={$titleText} />
        <Divider size={spacing.size12} type="vertical" />
        <Typography
          text={`${formatDate(new Date(item.createdDate).toISOString(), "yyyy-MM-dd HH:mm:ss")}`}
          preset="body02"
          style={$valueText}
        />
      </View>
    </Pressable>
  )
}

const $viewContainer: ViewStyle = {
  borderBottomWidth: 1,
  borderBottomColor: appColors.common.borderDivider,
  paddingVertical: spacing.size08,
  backgroundColor: appColors.common.bgWhite,
}

const $titleText: TextStyle = {
  minWidth: responsiveWidth(75),
}

const $valueText: TextStyle = {
  flexWrap: "wrap",
  overflow: "hidden",
}

const $rowItem: ViewStyle = {
  flex: 1,
  flexDirection: "row",
  overflow: "hidden",
}

const $logTypeView: ViewStyle = {
  position: "absolute",
  top: 0,
  right: 0,
}
