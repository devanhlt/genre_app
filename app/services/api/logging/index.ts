import { ApiResponse, ApisauceInstance, create } from "apisauce"
import { LoggingSnapshotIn } from "app/models/system/logging"

import Config from "../../../config"
import type { ApiConfig } from "../api.types"
import { GeneralApiProblem, getGeneralApiProblem } from "../apiProblem"
import { ApiSystemLoggingResponse, FilterLoggingPayload } from "./logging.api.types"

/**
 * Configuring the apisauce instance.
 */
export const DEFAULT_API_CONFIG: ApiConfig = {
  url: Config.API_URL,
  timeout: 10000,
}

/**
 * Manages all requests to the API. You can use this class to build out
 * various requests that you need to call from your backend API.
 */
export class LoggingApi {
  apisauce: ApisauceInstance
  config: ApiConfig

  /**
   * Set up our API instance. Keep this lightweight!
   */
  constructor(config: ApiConfig = DEFAULT_API_CONFIG) {
    this.config = config
    this.apisauce = create({
      baseURL: this.config.url,
      timeout: this.config.timeout,
      headers: {
        Accept: "application/json",
      },
    })
  }

  /**
   * Gets a list of system
   */
  async exportLoggingByFilter(
    filtering: FilterLoggingPayload,
  ): Promise<{ kind: "ok"; logging: LoggingSnapshotIn[] } | GeneralApiProblem> {
    // make the api call

    const response: ApiResponse<ApiSystemLoggingResponse> = await this.apisauce.post(
      `api/v1/system-logging/download/files`,
      { ...filtering },
    )

    // the typical ways to die when calling an api
    if (!response.ok) {
      const problem = getGeneralApiProblem(response)
      if (problem) return problem
    }

    try {
      const rawData = response?.data?.lstSystemLogging?.content

      // This is where we transform the data into the shape we expect for our MST model.
      const logging: LoggingSnapshotIn[] = rawData.map((raw) => ({
        ...raw,
      }))

      return { kind: "ok", logging }
    } catch (e) {
      if (__DEV__) {
        console.tron.error(`Bad data: ${e.message}\n${response.data}`, e.stack)
      }
      return { kind: "bad-data" }
    }
  }

  /**
   * Gets a list of system
   */
  async getLogging(
    filtering: FilterLoggingPayload,
  ): Promise<{ kind: "ok"; logging: LoggingSnapshotIn[] } | GeneralApiProblem> {
    // make the api call
    const response: ApiResponse<ApiSystemLoggingResponse> = await this.apisauce.get(
      `api/v1/system-logging/search?keyword=${filtering.keyword}&pageDraw=${
        filtering.pageDraw
      }&pageNumber=${filtering.pageNumber}&pageSize=${
        filtering.pageSize
      }&fromDate=${encodeURIComponent(
        filtering.fromDate.toISOString(),
      )}&toDate=${encodeURIComponent(filtering.toDate.toISOString())}`,
    )

    // the typical ways to die when calling an api
    if (!response.ok) {
      const problem = getGeneralApiProblem(response)
      if (problem) return problem
    }

    try {
      const rawData = response?.data?.lstSystemLogging?.content

      // This is where we transform the data into the shape we expect for our MST model.
      const logging: LoggingSnapshotIn[] = rawData.map((raw) => ({
        ...raw,
      }))

      return { kind: "ok", logging }
    } catch (e) {
      if (__DEV__) {
        console.tron.error(`Bad data: ${e.message}\n${response.data}`, e.stack)
      }
      return { kind: "bad-data" }
    }
  }
}

// Singleton instance of the API for convenience
export const apiLogging = new LoggingApi()
