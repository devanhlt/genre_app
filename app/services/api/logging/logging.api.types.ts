/**
 * These types indicate the shape of the data you expect to receive from your
 * API endpoint, assuming it's a JSON object like we have.
 */

import { Logging } from "app/models/system/logging"

export interface FilterLoggingPayload {
  keyword: string
  pageNumber: number
  pageSize: number
  pageDraw: number
  fromDate: Date
  toDate: Date
}

export interface ApiSystemLoggingResponse {
  lstSystemLogging: {
    content: Logging[]
  }
  totalCount: number
  recordsFiltered: number
  recordsTotal: number
}
