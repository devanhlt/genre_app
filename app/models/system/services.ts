import { ApiResponse } from "apisauce"

import { ApiServices } from "app/services/api"
import { GeneralApiProblem, getGeneralApiProblem } from "app/services/api/apiProblem"
import { Logging, LoggingFilterModel, LoggingSnapshotIn } from "./logging"
import { Instance } from "mobx-state-tree"
import { System } from "./system"

interface ApiSystemLoggingResponse {
  lstSystemLogging: {
    content: Logging[]
  }
  totalCount: number
  recordsFiltered: number
  recordsTotal: number
}

export default class SystemServices extends ApiServices {
  /**
   * Gets a list of system
   */
  async getSystems(): Promise<{ kind: "ok"; systems: System[] } | GeneralApiProblem> {
    // make the api call
    const timestamp = new Date().getTime()
    const response: ApiResponse<System[]> = await this.apisauce.get(`api/v1/systems?_=${timestamp}`)

    // the typical ways to die when calling an api
    if (!response.ok) {
      const problem = getGeneralApiProblem(response)
      if (problem) return problem
    }

    // transform the data into the format we are expecting
    try {
      const rawData = response.data

      // This is where we transform the data into the shape we expect for our MST model.
      const systems: System[] = rawData.map((raw) => ({
        ...raw,
      }))

      return { kind: "ok", systems }
    } catch (e) {
      if (__DEV__) {
        console.error(`Bad data: ${e.message}\n${response.data}`, e.stack)
      }
      return { kind: "bad-data" }
    }
  }

  /**
   * Export logging by current filter
   */
  async exportLoggingByFilter(
    filtering: Instance<typeof LoggingFilterModel>,
  ): Promise<{ kind: "ok"; logging: LoggingSnapshotIn[] } | GeneralApiProblem> {
    // make the api call

    const response: ApiResponse<Logging[]> = await this.apisauce.post(
      `api/v1/system-logging/download/files`,
      { ...filtering },
    )

    // the typical ways to die when calling an api
    if (!response.ok) {
      const problem = getGeneralApiProblem(response)
      if (problem) return problem
    }

    try {
      const rawData = response?.data

      // This is where we transform the data into the shape we expect for our MST model.
      const logging: LoggingSnapshotIn[] = rawData.map((raw) => ({
        ...raw,
      }))

      return { kind: "ok", logging }
    } catch (e) {
      if (__DEV__) {
        console.error(`Bad data: ${e.message}\n${response.data}`, e.stack)
      }
      return { kind: "bad-data" }
    }
  }

  /**
   * Export logging by current filter
   */
  async exportSingleLogDetail(
    logging: LoggingSnapshotIn,
  ): Promise<{ kind: "ok"; data: string } | GeneralApiProblem> {
    // make the api call

    const response: ApiResponse<string> = await this.apisauce.post(
      `api/v1/system-logging/download/file`,
      { ...logging },
    )

    // the typical ways to die when calling an api
    if (!response.ok) {
      const problem = getGeneralApiProblem(response)
      if (problem) return problem
    }

    try {
      const rawData = response?.data
      return { kind: "ok", data: rawData }
    } catch (e) {
      if (__DEV__) {
        console.error(`Bad data: ${e.message}\n${response.data}`, e.stack)
      }
      return { kind: "bad-data" }
    }
  }

  /**
   * Gets a list of system
   */
  async getLogging(
    filtering: Instance<typeof LoggingFilterModel>,
  ): Promise<{ kind: "ok"; logging: LoggingSnapshotIn[] } | GeneralApiProblem> {
    // make the api call
    const response: ApiResponse<ApiSystemLoggingResponse> = await this.apisauce.get(
      `api/v1/system-logging/search?keyword=${filtering.keyword}&pageDraw=${
        filtering.pageDraw
      }&pageNumber=${filtering.pageNumber}&logType=${filtering.logType}&system=${
        filtering.system ?? ""
      }&pageSize=${filtering.pageSize}&fromDate=${encodeURIComponent(
        filtering.fromDate.toISOString(),
      )}&toDate=${encodeURIComponent(filtering.toDate.toISOString())}`,
    )

    // the typical ways to die when calling an api
    if (!response.ok) {
      const problem = getGeneralApiProblem(response)
      if (problem) return problem
    }

    try {
      const rawData = response?.data?.lstSystemLogging?.content

      // This is where we transform the data into the shape we expect for our MST model.
      const logging: LoggingSnapshotIn[] = rawData.map((raw) => ({
        ...raw,
      }))

      return { kind: "ok", logging }
    } catch (e) {
      if (__DEV__) {
        console.tron.error(`Bad data: ${e.message}\n${response.data}`, e.stack)
      }
      return { kind: "bad-data" }
    }
  }
}
