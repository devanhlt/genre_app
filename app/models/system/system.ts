import { Instance, SnapshotIn, SnapshotOut, flow, toGenerator, types } from "mobx-state-tree"

import { withRootStore } from "../helpers/withRootStore"
import { Setting } from "../setting/setting"
import { withSetPropAction } from "../helpers/withSetPropAction"
import { apiSetting } from "app/services/api/setting"
import { jsonToString } from "app/utils/helpers"

export const SystemModel = types
  .model("System")
  .extend(withRootStore)
  .props({
    systemName: types.optional(types.maybeNull(types.string), null),
    baseUrl: types.optional(types.maybeNull(types.string), null),
    adminEmail: types.optional(types.maybeNull(types.string), null),
    adminPhone: types.optional(types.maybeNull(types.string), null),
    totalLog: types.optional(types.maybeNull(types.number), null),
    totalLogInfo: types.optional(types.maybeNull(types.number), null),
    totalLogWarn: types.optional(types.maybeNull(types.number), null),
    totalLogError: types.optional(types.maybeNull(types.number), null),
    receiveLog: types.optional(types.maybeNull(types.boolean), false),
  })
  .actions(withSetPropAction)
  .actions((self) => ({
    editSystemDetail: flow(function* editSystemDetail(system: Instance<typeof SystemModel>) {
      const response = yield* toGenerator(apiSetting.putSystemConfig(system))
      if (response.kind === "ok" && response.success === true) {
        self.setProp("systemName", system.systemName)
        self.setProp("baseUrl", system.baseUrl)
        self.setProp("adminEmail", system.adminEmail)
        self.setProp("adminPhone", system.adminPhone)
      } else {
        console.tron.error(`Error editing system: ${jsonToString(response)}`, [])
      }
    }),
    editSystemReceiveLog: flow(function* editSystemReceiveLog(
      system: Instance<typeof SystemModel>,
    ) {
      const response = yield* toGenerator(apiSetting.putSystemConfig(system))
      if (response.kind === "ok" && response.success === true) {
        self.setProp("receiveLog", system.receiveLog)
      } else {
        console.tron.error(`Error editing system: ${jsonToString(response)}`, [])
      }
    }),
  }))
  .views((self) => ({
    get getCurrentSystem() {
      return self
    },
    get getCurrentHeaderColor() {
      const { settingStore } = self.rootStore

      if (self.totalLogError > 0) {
        const settingLv2 = settingStore?.getCurrentSettings?.find((x: Setting) => x?.logLevel === 2)
        return settingLv2?.color || "#D32F2F"
      } else if (self.totalLogWarn > 0) {
        const settingLv1 = settingStore?.getCurrentSettings?.find((x: Setting) => x?.logLevel === 1)
        return settingLv1?.color || "#FBC02D"
      } else if (self.totalLogInfo > 0) {
        const settingLv0 = settingStore?.getCurrentSettings?.find((x: Setting) => x?.logLevel === 0)
        return settingLv0?.color || "#388E3C"
      } else if (self.totalLogInfo === 0 && self.totalLogError === 0 && self.totalLogWarn === 0) {
        return "#FBC02D"
      }
      return "#FFFFFF"
    },
  }))

export interface System extends Instance<typeof SystemModel> {}
export interface SystemSnapshotOut extends SnapshotOut<typeof SystemModel> {}
export interface SystemSnapshotIn extends SnapshotIn<typeof SystemModel> {}
