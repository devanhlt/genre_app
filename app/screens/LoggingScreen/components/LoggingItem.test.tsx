import React from "react"

import { Logging } from "app/models/system/logging"
import renderer from "react-test-renderer"
import LoggingItem from "./LoggingItem"

describe("<UserItem />", () => {
  it("renders correctly when item data is correctly", () => {
    const user = {
      id: "1",
      systemName: "EKYC",
      apiReferal: "Test",
      logType: "Test",
      logCode: 1,
      logLevel: 2,
      logDetail: "Test",
      method: "POST",
      requestParam: "Test",
      requestBody: "Test",
      startTime: "Test",
      endTime: "Test",
      status: "Test",
      createdBy: "Test",
      createdDate: 1,
      updatedBy: "Test",
      updatedDate: 1,
      idStr: "Test",
      color: "red",
    } as Logging
    const tree = renderer.create(<LoggingItem item={user} />)
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly when item is empty", () => {
    const tree = renderer.create(<LoggingItem item={null} />)
    expect(tree).toMatchSnapshot()
  })
})
