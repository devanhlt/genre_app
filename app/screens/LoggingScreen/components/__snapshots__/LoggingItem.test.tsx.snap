// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserItem /> renders correctly when item data is correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      [
        {
          "paddingHorizontal": 16,
          "paddingVertical": 12,
        },
        {
          "backgroundColor": "#C5281C",
          "paddingHorizontal": 12,
          "paddingVertical": 12,
        },
      ],
      {
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#E5E8ED",
        "borderBottomWidth": 1,
        "paddingVertical": 8,
      },
      false,
      false,
    ]
  }
>
  <View
    style={
      [
        [
          {
            "flexGrow": 0,
            "flexShrink": 1,
            "marginStart": 4,
            "textAlign": "center",
            "zIndex": 2,
          },
          {
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "600",
            "letterSpacing": 0.15,
            "lineHeight": 20,
          },
          {
            "color": "#FFFFFF",
          },
        ],
        undefined,
        false,
        false,
      ]
    }
  >
    <View
      style={
        {
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Text
        collapsable={false}
        numberOfLines={1}
        style={
          {
            "alignSelf": "flex-end",
            "backgroundColor": "red",
            "borderRadius": 4,
            "color": "#fff",
            "fontSize": 10,
            "height": 20,
            "lineHeight": 10,
            "minWidth": 20,
            "opacity": 1,
            "overflow": "hidden",
            "paddingHorizontal": 4,
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
      >
        Test
      </Text>
    </View>
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <Text
        preset="body04"
        style={
          [
            {},
            {
              "color": "#000F27",
            },
            {
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "600",
              "letterSpacing": 0.15,
              "lineHeight": 24,
            },
            {
              "minWidth": 75,
            },
          ]
        }
      >
        System:
      </Text>
      <View
        style={
          [
            {
              "flexGrow": 0,
              "flexShrink": 0,
            },
            false,
            {
              "width": 12,
            },
            undefined,
          ]
        }
      />
      <Text
        preset="body04"
        style={
          [
            {},
            {
              "color": "#000F27",
            },
            {
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "600",
              "letterSpacing": 0.15,
              "lineHeight": 24,
            },
            {
              "flexWrap": "wrap",
              "overflow": "hidden",
            },
          ]
        }
      >
        EKYC
      </Text>
    </View>
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <Text
        preset="body04"
        style={
          [
            {},
            {
              "color": "#000F27",
            },
            {
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "600",
              "letterSpacing": 0.15,
              "lineHeight": 24,
            },
            {
              "minWidth": 75,
            },
          ]
        }
      >
        Event time:
      </Text>
      <View
        style={
          [
            {
              "flexGrow": 0,
              "flexShrink": 0,
            },
            false,
            {
              "width": 12,
            },
            undefined,
          ]
        }
      />
      <Text
        preset="body02"
        style={
          [
            {},
            {
              "color": "#000F27",
            },
            {
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
              "letterSpacing": 0.15,
              "lineHeight": 24,
            },
            {
              "flexWrap": "wrap",
              "overflow": "hidden",
            },
          ]
        }
      >
        1970-01-01 08:00:00
      </Text>
    </View>
  </View>
</View>
`;

exports[`<UserItem /> renders correctly when item is empty 1`] = `null`;
